tutorials/15-external-samplers/external-samplers.jmd

```diff
@@ -0,0 +1,104 @@
+In order to avoid type ambiguity within Turing, at the moment is necessary to declare `spl` as an external sampler to Turing this is done as:
+
+```julia
+espl = externalsampler(spl)
+```
+
+where `externalsampler(s::AbstractMCMC.AbstractSampler)` is a Turing function that types our external sampler adquately.
+
+Arrived at this point sampling our model is as easy as:
+
+```julia
+chain = sample(model, espl, 1_000)
+```
+
+This process will use `AbstractMCMC` functions overloaded by Turing to generate chains identical to those generated by the native samplers.
+Note that if the external sampler produces transitions that Turing cannot parse the bundling of the samples will be different or fail.
+
+## AdvancedMH
+
+A good point to start to show how this is done in practice is by looking at the sampling library `AdvancedMH` for Metropolis Hastings methods.
+Let's say that we want to use a random walk Metropolis Hastings sampler without specifying the proposal distributions.
+We can do this in two different ways:
+
+```julia
+# Importing the sampling library
+using AdvancedMH
+
+rwmh_1 = RWMH(d)
+rwmh_2 = RWMH(model)
+```
+
+The first method uses a multivariate Gaussian distribution with zero mean and unit variance in `d` dimensions.
+The second method uses the priors of the model as a proposal. Note that the second method requires Turing to be imported.
+
+Let us now compare the performance and posteriors of these two samplers to the native MH sampler within Turing.
+
+```julia
+mh_chain_0 = sample(model, MH(), 1_000)
+mh_chain_1 = sample(model, externalsampler(rwmh_1), 1_000)
+mh_chain_2 = sample(model, externalsampler(rwmh_2), 1_000)
+```
+
+## AdvancedHMC
+
+As previously mentioned the Turing wrappers can often limit the capabilities of the sampling libraries they wrap.
+AdvancedHMC is a clear example of this. A common practice when performing HMC is to provide an initial guess for the mass matrix.
+However, the native HMC sampler within Turing only allows the user to specify the type of the mass matrix despite the two options being possible within AdvancedHMC.
+Thankfully, we can use Turing's support for external samplers to define a HMC sampler with a custom mass matrix in AdvancedHMC and then use it to sample our Turing model.
+
+## Micro-Canonical HMC
+
+So far we have used to Turing's support for external samplers to go beyond the capabilities of the wrappers.
+Now we want to use this support to employ a sampler not supported within Turing's ecosystem yet.
+We will use the recently developped Micro-Cannoncial Hamiltonian Monte Carlo (MCHMC) sampler to showcase this.
+MCHMC[^1,^2] is HMC sampler that conserves energy along the exploration of the phase space.
+This is achieved by simulating the dynamics of a microcanonical Hamiltonian with an additional noise term to ensure ergodicity.
+The algorithm has been shown to outperform traditioanl HMC in many scenarios and it doesn't require of Metropolis adjustments.
```

Comment:
IMO this is a bit too salespitchy :sweat_smile: 

> The algorithm has been shown to outperform traditioanl HMC

I'd prefer to not make statements like this unless we're also significantly backing these claims up. People often look to PPL docs for what's considered state-of-the-art, and so if they see this, they might think this is an endorsement of micro-canonical HMC, which I'm not quite comfortable with _at this moment_  :wink: 

> it doesn't require of Metropolis adjustments.

This is stated as a feature, but there are tons of methods (also already present in Turing, e.g. Langevin)  which does not require this, _but_ they (almost all) come at the cost of not being asymptotically (in the number of iterations) unbiased; IIUC this is also the case for micro-canonical HMC.

Reply:
Fair enough! I will make it more decaff.